### DISCUSSION TITLE
Edit record

### DISCUSSION THREAD

**Discussion by JoostvanBruggen** [Mattk70/Chirpity-Electron#96] [General]
When I change a record-ID from Little Grbe to Whimbrel by right clicking on the record. Then opt for "Edit Record" and change it to Whimbrel. Do I change it only for me or do I help you with further enhancement and beter results of Chirpity?

**Comment by Mattk70** [comment]
You just change it for you. 

I'm thinking about how to best have Chirpity learn from corrections. Whilst this would be possible, it is far from strighforward. There are two approachies I am considering, for differrent use caes. The first is to have an option to share mis-identified audio recordings with me, so I can use these to develop more robust models for everyone. The second is to allow individuals to train and use custom models, for instance - to help more accurately identify specific species' call types, as this would be useful to people like Lars Gunnarn: https://github.com/Mattk70/Chirpity-Electron/issues/86 

In the latter case, and indeed with all cases, a good selection of training data is needed to improve the models. It is not a case of the model incrementally improving wth each correction.

**Reply by JoostvanBruggen** [reply]
Would you like to receive nocturnal recordings from me? (complete nights)

**Reply by Mattk70** [reply]
Not as a way of collecting more training data, no. What I had done in very early versions of the application was to give an option for people to upload a 3-second detection that was incorrectly identified. I could re-enable this, but I'd need to think about how to manage the collection of what could end up being a lot of data. Would you use such a feature?

Where the misidentification is a false negative (it's not picked up a genuine call) my strong preference is to enable an "upload to XC" feature, such that the audio can be put on Xeno-canto and benefit all. This needs the guys at XC to provide an upload API. If that's something that interests you, please go ahead and say so on this article: https://xeno-canto.org/article/287. [Willem-Pier Vellinga](https://xeno-canto.org/contributor/AEFZWKWLMO) suggested this may happen but I have not heard anything from him about it since.

**Reply by JoostvanBruggen** [reply]
I know Willem-Pier. I consider him a friend. I'll contact him!

**Reply by Vollpflock** [reply]
Thank you very much, Joost!

**Reply by Mattk70** [reply]
Yes, thank you @JoostvanBruggen - it really would be a great feature to add.

**Reply by JoostvanBruggen** [reply]
Willem is going to respond on Xeno-Canto to you!

**Reply by Mattk70** [reply]
Awesome, thank you for getting in touch with him, Joost
